src/Http/WebUtilities/test/MultipartReaderTests.cs

```diff
@@ -315,6 +334,57 @@ public void MultipartReader_BufferSizeMustBeLargerThanBoundary_Throws()
         });
     }
 
+    [Fact]
+    public async Task MultipartReader_ReadMultipartBodyWithFilesForDeferredCopy_Success()
+    {
+        var stream = MakeStream(ThreePartBodyWithMixedFiles);
+        var reader = new MultipartReader(Boundary, stream);
+
+        var section = await reader.ReadNextSectionAsync();
+        Assert.NotNull(section);
+        Assert.Equal(2, section.Headers.Count);
+        Assert.Equal("form-data; name=\"file1\"; filename=\"a.txt\"", section.Headers["Content-Disposition"][0]);
+        Assert.Equal("text/plain", section.Headers["Content-Type"][0]);
+        var stream1 = section.Body;
+
+        section = await reader.ReadNextSectionAsync();
+        Assert.NotNull(section);
+        Assert.Equal(2, section.Headers.Count);
+        Assert.Equal("form-data; name=\"file2\"; filename=\"a.html\"", section.Headers["Content-Disposition"][0]);
+        Assert.Equal("text/html", section.Headers["Content-Type"][0]);
+        var stream2 = section.Body;
+
+        section = await reader.ReadNextSectionAsync();
+        Assert.NotNull(section);
+        Assert.Equal(2, section.Headers.Count);
+        Assert.Equal("form-data; name=\"file3\"; filename=\"a.json\"", section.Headers["Content-Disposition"][0]);
+        Assert.Equal("application/json", section.Headers["Content-Type"][0]);
+        var stream3 = section.Body;
+
+        Assert.Null(await reader.ReadNextSectionAsync());
+
+        Assert.True(stream1.CanSeek);
+        Assert.Equal(0, stream1.Seek(0, SeekOrigin.Begin));
+        var buffer = new MemoryStream();
+        var exception = await Record.ExceptionAsync(() => stream1.CopyToAsync(buffer));
+        Assert.Null(exception);
```

Comment:
```suggestion
        await stream1.CopyToAsync(buffer);
```

There's no need to null check exceptions. If `CopyToAsync` throws, it will fail the test already. I'd remove the other two calls to `Record.ExceptionAsync` too. Thanks.

Reply:
@halter73 : Ok, good suggestion. it should be solved. Thanks!